### PR TITLE
WIP Entity menu ellipsis

### PR DESCRIPTION
**Pull request by hypeJunction**
entity

### CODE REVIEW

**Comment by hypeJunction** [comment]
We can do this for objects right away. With users, we can decide whether we keep the user hover menu or not. I'd rather we get rid of it and start consolidating around entity menu for all actions.

File: views/default/navigation/menu/entity.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+unset($vars['menu']['default']);
+
+if (!empty($vars['menu'])) {
+	$id = "elgg-popup-" . base_convert(mt_rand(), 10, 36);
+	$link = elgg_view('output/url', [
+		'rel' => 'popup',
+		'text' => elgg_view_icon('ellipsis-v'),
+		'href' => "#{$id}",
+		'class' => 'elgg-popup-inline',
+		'data-position' => json_encode([
```
Comment (hypeJunction) [inline views/default/navigation/menu/entity.php]: This should go in a JS hook.

**Comment by jdalsem** [comment]
are these menus ajax loaded? or is that now gone...

**Comment by hypeJunction** [comment]
Now ajax

**Comment by hypeJunction** [comment]
No more ajax. Can be added I suppose but that doesn't make sense for all entity menus

**Comment by jdalsem** [comment]
when we moved the userhover menu to an ajax hook this made a significant improvement ... would not like to see us go back a few years.. why would it not apply to all entity menus?

**Comment by hypeJunction** [comment]
Because they only have a few items. We can make it ajax but it's a whole new level of complexity. This is just a demo.  We don't have to move uset hover yet.

**Comment by jdalsem** [comment]
> Because they only have a few items.

maybe for now...



> We can make it ajax but it's a whole new level of complexity. This is just a demo. We don't have to move uset hover yet.

it makes sense to do a similar ellipsis for user hover

potential downside of hidden menus is the fact that the will never be used, but still require server processing. Show 50 entities with 5 menu items with all the hooks... that will take some time. It was for users anyway... entities will be a lot less, but the ellipsis menu will encourage usage of the entity menu, so we should prepare for that

**Comment by jdalsem** [comment]
i like where this is going btw... 

does this mean *all* entity menu items are in the ellipsis? i was wondering if we could separate primary (always visible) and secondary (in the ellipsis) menu items

You now set different colors for various sections:
i was wondering if we could create some default styled/colored sections for 
- generic actions
 - send message
 - add friend
- owner actions
 - edit content
- group admin actions
 - kick from group
 - make group admin
- admin actions
 - delete user
 - ban

**Comment by jdalsem** [comment]
> does this mean all entity menu items are in the ellipsis? i was wondering if we could separate primary (always visible) and secondary (in the ellipsis) menu items

is see you created a discussion for that here: #10420

**Comment by jdalsem** [comment]
maybe you could also take a look at #6936 while you are at it

**Comment by hypeJunction** [comment]
There is also this bugger on the way #9126.

**Comment by jdalsem** [comment]
there are always items that serve both an informational purpose and also have an action, like the like/comment actions + counts or the access information (combined with plugins like this https://github.com/hypeJunction/Elgg-menus_access)... those menu items should not be part of the ellipsis. I would opt to make entity menu items by default be part of the ellipsis section and if you need it elsewhere, you register for an other section, that will just show inline (like it does now)

**Comment by hypeJunction** [comment]
We can't really make this work with ajax without breaking the menu down into different menus (no way of telling when the hook should be adding items). Perhaps we could keep user hover as a menu loaded via ajax but move the trigger to the entity menu. And add similar menus for objects and groups. I am letting this site for a bit until I there is some clear roadmap

**Comment by jeabakker** [comment]
if you come back to this, maybe also look at #6402

**Comment by hypeJunction** [comment]
I have a different approach in mind. Will resubmit in another PR

**Comment by hypeJunction** [comment]
Strawman:

- Teach `elgg/popup` to load ajax views as we do in tabs
- Add `more` menu item that links to the appropriate view with the ellipsis menu (rename `user_hover` to `entity:more`)
- Render `entity` menu regardless of context, inject all view vars to entity menu, let plugins decide when to render menu items

**Comment by hypeJunction** [comment]
Another PR will be coming eventually.

**Pull request closed without merging**